shared/AppInsightsCore/src/JavaScriptSDK/SenderPostManager.ts

```diff
@@ -135,6 +135,72 @@ export class SenderPostManager {
                 _initDefaults();
             };
 
+            _self.preparePayload = (callback: (processedPayload: IPayloadData) => void, disableZip: boolean, payload: IPayloadData, isSync: boolean) => {
+                if (disableZip || isSync || !payload.data) {
+                    // If the request is synchronous, the body is null or undefined or Compression is not supported, we don't need to compress it
+                    callback(payload);
+                    return;
+                }
+
+                const CompressionStream = (window as any).CompressionStream;
```

Comment:
We should also wrap all of this in a try / catch and call the callback(payload) on error to ensure that if something here fails we still send the payload.